File: tenk-backend/src/main/java/com/hjson/tenk/domain/auth/AuthTokens.java
package com.hjson.tenk.domain.auth;

public record AuthTokens(
        String accessToken,
        String refreshToken,
        long accessTokenExpiresIn,
        Long userId,
        String nickname,
        // 이번 카카오 로그인이 신규 가입을 만든 경우 true. refresh 응답에서는 항상 false.
        // 클라이언트는 true 일 때 NicknameSetupScreen 으로 분기 (가입 화면 강제).
        boolean isNewUser,
        // 필수 동의(이용약관 + 개인정보 수집·이용) 미완료 여부. 로그인 직후 클라이언트가 동의 게이트 분기에 사용.
        // 신규 가입은 항상 true, 테스트 계정은 auto-consent 라 false.
        boolean consentRequired
) {}

File: tenk_app/lib/data/auth/auth_tokens.dart
/// 백엔드가 발급한 자체 JWT 한 쌍.
///
/// [isNewUser] 는 카카오 로그인 응답에서만 의미 있음 (이번 호출이 신규 가입을 만들었는지).
/// refresh 응답이나 storage 에서 복원한 토큰은 항상 false. 저장되지도 않는다.
class AuthTokens {
  AuthTokens({
    required this.accessToken,
    required this.refreshToken,
    this.isNewUser = false,
    this.consentRequired = false,
  });

  final String accessToken;
  final String refreshToken;
  final bool isNewUser;

  /// 필수 동의 미완료 여부. 로그인 응답에서만 의미 있음 (storage 복원 토큰은 항상 false).
  /// 로그인 직후 동의 게이트 분기에 사용.
  final bool consentRequired;

  factory AuthTokens.fromJson(Map<String, dynamic> json) => AuthTokens(
        accessToken: json['accessToken'] as String,
        refreshToken: json['refreshToken'] as String,
        isNewUser: json['isNewUser'] as bool? ?? false,
        consentRequired: json['consentRequired'] as bool? ?? false,
      );
}
